common/models/acl.js

```diff
@@ -456,15 +489,34 @@ module.exports = function(ACL) {
         if (err) return callback(err, null);
 
         var resolved = self.resolvePermission(effectiveACLs, req);
+        var permission = resolved && resolved.permission;
         if (resolved && resolved.permission === ACL.DEFAULT) {
-          resolved.permission = (model && model.settings.defaultPermission) || ACL.ALLOW;
+          permission = defaultPermission || ACL.ALLOW;
         }
         debug('---Resolved---');
         resolved.debug();
+
+        // set authorizedRoles in remotingContext options argument if
+        // checkAccessForContext resolved permission status is ALLOW,
+        // else set it to empty object
+        authorizedRoles = permission === ACL.ALLOW ? authorizedRoles : {};
+        setAuthorizedRoles(authorizedRoles);
```

Comment:
reduced down the whole setAuthorizedRoles logic to just these 2 lines in the main code block